src/commands/time/finish.js

```diff
@@ -24,103 +24,120 @@ function controller(t) {
-          utils.log.chalk('red', '    System is configured NOT to log bug times AT ALL.');
-          return done();
-        }
+        getResource.then(function (result) {
+          tp.getIssueTimeTo(result.Project.Id).then(function(value) {
+            console.log(result.Project.Id);
+            var issueTimeToValue = value.items[0].issueCFRaw;
 
-        var getter;
-        if (isBug) {
-            getter = tp.getBug;
-        } else if (isTask) {
-            getter = tp.getTask;
-        } else {
-            getter = tp.getStory;
-        }
-        getter.call(tp, e.tp_task ? e.tp_task.id : e.tp_user_story.id)
-        .then(function (tpTask) {
-            base.captureTimeRemaining(e.hours, tpTask, function (remain) {
-
-                var tpdata = {
-                    description: e.notes || '-',
-                    spent: e.hours,
-                    remain: remain,
-                    date: new Date(e.created_at).toJSON()
-                };
-
-                function logTime_us(cb) {
-                  // bugs may be without user-story
-                  var isUserStory = tpTask.ResourceType === 'UserStory';
-                  if(!isUserStory && !tpTask.UserStory) {
-                    utils.log.chalk('red', '    This '+tpTask.ResourceType+' is not associated with a user-story. -- ignored');
-                    return cb();
-                  }
-
-                  var userStoryId = isUserStory ? tpTask.Id : tpTask.UserStory.Id;
-                  
-                  var tpusdata;
-                  if (isUserStory) {
-                    utils.log('    Logging time on the user story...');
-                    tpusdata = {
-                        description: 'time spent on user story #' + userStoryId,
-                        spent: e.hours,
-                        date: new Date(e.created_at).toJSON()
-                    };
-                  } else {
-                    utils.log('    Logging bug time on the user story...');
-                    tpusdata = {
-                        description: 'time spent on bug #' + e.tp_task.id,
+            // configured to not log bug-time at all
+            if(isBug && issueTimeToValue === 'none') {
+              utils.log.chalk('red', '    System is configured NOT to log bug times AT ALL.');
+              return done();
+            }
+
+            var getter;
+            if (isBug) {
+                getter = tp.getBug;
+                utils.log('    Bugs are configured to be logged against: ' + issueTimeToValue);
+            } else if (isTask) {
+                getter = tp.getTask;
+                utils.log('    Bugs are configured to be logged against: ' + issueTimeToValue);
+            } else {
+                getter = tp.getStory;
+                utils.log('    Bugs are configured to be logged against: ' + issueTimeToValue);
+            }
+
+            getter.call(tp, e.tp_task ? e.tp_task.id : e.tp_user_story.id)
```

Comment:
We can actually remove this from back to line 50. We no-longer need set `getter` and `call` it, since `tpTask` should end up being the same as `result` (the TP entity that the user is logging time to).